<launch>

  <arg name="a" default="0.2"/>
  <arg name="b" default="0.4"/>
  <arg name="c" default="0.08"/>

  <arg name="r" default="0.05"/>
  <arg name="w_max" default="10.0"/>

  <arg name="x_spacing" default="0.2"/>
  <arg name="y_spacing" default="0.2"/>

  <arg name="sensor_right_x" default="0.0"/>
  <arg name="sensor_right_y" default="-0.2"/>

  <arg name="sensor_left_x" default="0.0"/>
  <arg name="sensor_left_y" default="0.2"/>

  <arg name="file_name" default="you_forgot_to_put_a_name"/>

  <!-- joy node -->
  <node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
    <param name="dev" type="string" value="/dev/input/js0" />
  </node>

  <!-- The compute_twist adapter -->
  <node respawn="true" pkg="simulation" type="simulation_compute_twist" name="compute_twist" output="screen">
  </node>

  <node pkg="rosbag" type="record" name="simulation_recorder" args="--output-name=$(find data)/recorded_data/$(arg file_name) /simulation/EncodersReading /simulation/IRSensorsStatus /simulation/RobotPosture">
  </node>

  <group ns="simulation">
    <!-- The robot_2_0 -->
    <node pkg="simulation" type="simulation_robot_2_0" name="robot_2_0" output="screen">
      <param name="track_gauge"         value="$(arg a)"      />
      <param name="wheel_radius"        value="$(arg r)"      />
      <param name="joint_offset"        value="$(arg b)"      />
      <param name="castor_arm"          value="$(arg c)"      />
      <param name="actuator_max_speed"  value="$(arg w_max)"  />
    </node>

    <!-- The sensor server
    <node pkg="simulation" type="simulation_sensor_server" name="sensor_server" output="screen">
      <param name="x1_pos" value="$(arg sensor_right_x)" />
      <param name="y1_pos" value="$(arg sensor_right_y)" />
      <param name="x2_pos" value="$(arg sensor_left_x)" />
      <param name="y2_pos" value="$(arg sensor_left_y)" />
      <param name="x_spacing" value="$(arg x_spacing)" />
      <param name="y_spacing" value="$(arg y_spacing)" />
    </node> -->

    <!-- The sensor server -->
    <node pkg="simulation" type="simulation_sensor" name="sensor" output="screen">
      <param name="x1_pos" value="$(arg sensor_right_x)" />
      <param name="y1_pos" value="$(arg sensor_right_y)" />
      <param name="x2_pos" value="$(arg sensor_left_x)" />
      <param name="y2_pos" value="$(arg sensor_left_y)" />
      <param name="x_spacing" value="$(arg x_spacing)" />
      <param name="y_spacing" value="$(arg y_spacing)" />
    </node>

    <!-- Launch Rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find simulation)/rviz/first_save.rviz">
    </node>

  </group>

  <!-- Save the parameters -->
  <rosparam command="dump" file="$(find data)/recorded_data/$(arg file_name).yaml"  />




</launch>
